### PR TITLE
Give the repository a base structure

### PR DESCRIPTION
**Pull request by sgametrio**
This PR tries to give structure and order to facilitate reasoning about templates and how to use them. It enables having templates as folders and not single files.

I identified, as of now, two kind of templates:
1. Bases
2. Addons

bases are meant to initialize a new repository.

addons are mant to be used as addons (heh) to create changes to existing repositories.

The folder structure helps the usage of them too (from the Tam CLI).

To validate the usefulness of the structure, I added 2 template:
- sfmc new sample app: Base template to bootstrap a new react app. Stops at pushing image to ECR as afterwards the deployment is meant to be managed by Flux
- flux-new-service: Addon template to add deployment to the flux-infra repository for a new service

used by https://github.com/DND-IT/tam-cli/pull/44

### Breaking changes

1. Move `jinja/lb.tf.tmpl` to `addons/lb/<name>.tf`

### Follow up actions

1. Move `template-generic-app` to the `base` folder

### CODE REVIEW

File: addons/flux-new-service/apps/dev/<service_name>/app-SealedSecret-env.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# This file should be created through devopstools UI
```
Comment (AlfGot) [inline addons/flux-new-service/apps/dev/<service_name>/app-SealedSecret-env.yaml:1]: @sgametrio devops tool repo update and the sealed secret creation are manually managed?

File: addons/flux-new-service/apps/dev/<service_name>/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+  - ../../base/@{{ service_name }}
+  # - app-SealedSecret-env.yaml
```
Comment (AlfGot) [inline addons/flux-new-service/apps/dev/<service_name>/kustomization.yaml:5]: Is this something we handle in a second step, once the sealed secret has been possibly created, correct? Does it mean that the app deployment will fail till the selaed secret is created, correct?

**Comment by AlfGot** [comment]
@sgametrio I like the approach and the structure, I've just asked a couple of questions. How can we validate the template works as expected?

**Comment by sgametrio** [comment]
@AlfGot Thanks for the comments. The scope of the added templates is to automate **part of** the request that we receive when a new service is needed. There are still some manual steps to be done. 

I summarized them in a comment in the closed ticket - https://tamedia.atlassian.net/browse/FUM-3243?focusedCommentId=324289

**Comment by AlfGot** [comment]
> I summarized them in a comment in the closed ticket - https://tamedia.atlassian.net/browse/FUM-3243?focusedCommentId=324289

This is great! How could we make this information easier to remember or find? Would it make sense to create a sort of _readme file_ or _how to_ for each template we create, or at least the ones where we believe valuable to add son extra info?

**Comment by sgametrio** [comment]
@AlfGot Linked in the right spot: https://tamedia.atlassian.net/wiki/x/0A2kAg 

This is where I took the procedure

**Review by AlfGot — COMMENTED** [review]
For me it's ok and I could approve it, but it might be worth to get a few further feedbacks

**Comment by sgametrio** [comment]
I will merge this so that I unlock tam-cli#44. Anyway the LB is not in use at the moment. 

For further feedback, we can create new PRs afterwards.

**Pull request merged**